File: myproj26/main01.py
name = "박예린"  # 새로운 변수 선언
name = "Merryno"  # 변겅

number = 10

if number % 2 == 0:
    print("짝수")
else:
    print("홀수")


for i in range(1, 11):
    print(i)


for i in range(1, 11, 2):
    print(i)


# 함수

def mysum(x, y):
    return x+y

mysum2 = lambda x, y: x + y


def mysum5(x, y, *args):
    return x + y + sum(args)


print(mysum(1, 2, 3, 4, 5))

File: myproj26/main01.js
// var name = "박예린"
// name = "Merryno"


// 변수
let name = "박예린";
name = "Merryno";


// 상수
const age = 10;
// age = 12; // 상수는 값 변경 불가


console.log("name, age")


const number = 10;

if (number % 2 === 0) {
    console.log("짝수");
}
else {
    console.log("홀수")
}


for (let i = 1; i < 11; i++) {
    console.log(i);
}

for (let i = 1; i < 11; i += 2) {
    console.log(i);
}


// 함수

function mysum(x, y) {
    return x + y;
}

const mysum2 = function (x, y) {
    return x + y;
};

const mysum3 = (x, y) => {
    return x + y;
}


const mysum4 = (x, y) => x + y;


function mysum5(x, y, ...args) {
    console.log(x, y, args);
}

mysum5(1, 2, 3, 4, 5)

function reducer(prevValue, currentValue) {
    return prevValue + currentValue;
}

const result = [1, 2, 3, 4, 5].reduce(reducer, 0);
console.log(result);

const result2 = [1, 2, 3, 4, 5].reduce((prevValue, currentValue) => {
    return prevValue + currentValue;
}, 0);
console.log(result2);


const result3 = [1, 2, 3, 4, 5].reduce(
    (prevValue, currentValue) => prevValue + currentValue,
    0);
console.log(result3);

